src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityRootAction.java

```diff
@@ -568,18 +571,62 @@ private synchronized void load() {
             LOGGER.log(Level.SEVERE, "Failed to load the configuration from config path = "+config,e);
         }
     }
-    
+
+    /**
+     * Represents the different response formats (JSON, pretty JSON, XML).
+     */
+    private enum ResponseFormat {
+        JSON,
+        PRETTYJSON,
+        XML;
+
+        private static final ResponseFormat DEFAULT = JSON;
+
+        public static ResponseFormat fromString (final String format) {
+            if (format == null) {
+                return DEFAULT;
+            }
+            if (format.equals("json")) {
+                return JSON;
+            } else if (format.equals("json-pretty")) {
+                return PRETTYJSON;
+            } else if (format.equals("xml")) {
+                 // Related to https://issues.jenkins-ci.org/browse/JENKINS-28727
+                return DEFAULT;
+            } else {
+                return DEFAULT;
+            }
+        }
+    }
+
     /**
-     * Serves the JSON response.
-     * @param item Data to be serialized to JSON
-     * @return HTTP response with application/json MIME type
+     * Serves the response and manages its output format in the response.
+     *
+     * @param item Data to be serialized
+     * @param format Format used in the response
+     *
+     * @return HTTP response with MIME type
      */
-    private static HttpResponse toJSONResponse(final Object item) {
+    private static HttpResponse toFormattedResponse(final Object item, final String format) {
         return new HttpResponse() {
             @Override
             public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node) throws IOException, ServletException {
-                ObjectMapper mapper = new ObjectMapper(); 
-                rsp.setContentType("application/json;charset=UTF-8");
+                ObjectMapper mapper = new ObjectMapper();
+                ResponseFormat responseFormat = ResponseFormat.fromString(format);
+                switch (responseFormat) {
+                    case JSON:
+                        rsp.setContentType("application/json;charset=UTF-8");
+                        break;
+                    case PRETTYJSON:
+                        mapper.enable(SerializationFeature.INDENT_OUTPUT);
+                        rsp.setContentType("application/json;charset=UTF-8");
+                    case XML:
+                        // Related to https://issues.jenkins-ci.org/browse/JENKINS-28727
+                        break;
+                    default:
+                        rsp.setContentType("application/json;charset=UTF-8");
```

Comment:
It should not happen according to your code. I would throw an exception in order to let developers know about the error